scripts/zones/Yhoator_Jungle/npcs/qm3.lua

```diff
@@ -15,7 +15,8 @@ end
 
 function onTrigger(player,npc)
     if player:getQuestStatus(OUTLANDS, dsp.quest.id.outlands.TRUE_WILL) == QUEST_ACCEPTED and not player:hasKeyItem(dsp.ki.OLD_TRICK_BOX) then
-        if player:getCharVar("trueWillKilledNM") > 0 then
+        local mobsStillUp = GetMobByID(ID.mob.KAPPA_AKUSO):isSpawned() or GetMobByID(ID.mob.KAPPA_BONZE):isSpawned() or GetMobByID(ID.mob.KAPPA_BIWA)
```

Comment:
This is still just attempting to confirm all 3 are not currently up.

There is zero way the npc can tell if the mob was killed or just depopped, also that last one in the var just checks if the mob exists. basically whats needs to happen is checking isSpawned at time of death on the others - whether they are dead or depopped won't matter, so long as its the last of the 3 to be killed by the player. the scripts you changed in the 1st pr mistakenly check on nil entities they had not defined and was checking the wrong state (I think - maybe a despawned mob still counts as "dead" tho). That author was using for `KAPPA_AKUSO` instead of `GetMobByID(ID.mob.KAPPA_AKUSO)` which is why it was nil.

Reply:
nvm jithoob was showing me old version I had to refresh sorry